File: 04_quicksort/swift/04_recursive_max.swift
import Foundation

func max(list : [Int]) -> Int {
    if list.count == 2 {
        return (list[0] > list[1]) ? list[0] : list[1]
    } else if list.count < 2 {
        return list.first ?? 0
    }
    var tempArray = list
    tempArray.remove(at: 0)
    let subMax = max(list: tempArray)
    return (list[0] > subMax) ? list[0] : subMax
}

print(max(list: [1, 5, 10, 25, 16, 1])) // => 25

File: 04_quicksort/scala/04_recursive_max.scala
def max(list: List[Int]): Int = {
  if(list.length == 2){
    if(list(0) > list(1)) return list(0) 
    else return list(1)
  }
  val sub_max = max(list.drop(1))
  if(list(0) > sub_max) list(0) 
  else sub_max
}